hooks/caret-position.js

```diff
@@ -0,0 +1,34 @@
+
+import {useState, useRef, useEffect, useCallback} from 'react'
+
+import useFocus from './focus'
+
+function useCaretPosition({initialValue, isFocus}) {
+  const [focusedElement] = useFocus({value: null, isFocus})
+  const value = useRef(initialValue) // Save initial input value state on first render
+
+  const [start, setStart] = useState(value.current ? value.current.length : 0)
+  const [end, setEnd] = useState(value.current ? value.current.length : 0)
```

Comment:
Il n'est pas nécessaire de créer deux states. Les deux étant dépendantes l'une de l'autre au moment de la mise à jour, nous pouvons les regrouper.

```suggestion
  const initialCursorPos = value.current ? value.current.length : 0
  const [cursorPos, setCursorPos] = useState({start: initialCursorPos, end: initialCursorPos})
```